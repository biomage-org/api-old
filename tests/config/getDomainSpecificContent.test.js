const getDomainSpecificContent = require('../../src/config/getDomainSpecificContent');
const { ACCOUNT_ID } = require('../../src/api.v2/constants');

describe('getDomainSpecificContent tests', () => {
  // Make a snapshot for each account
  Object.keys(ACCOUNT_ID).forEach((accountId) => {
    test(`check content for ${accountId}`, () => {
      // Set the environment variables
      process.env.AWS_ACCOUNT_ID = ACCOUNT_ID[accountId];
      process.env.NODE_ENV = 'production'; // Since the function checks if env is not 'test'

      const content = getDomainSpecificContent();
      expect(content).toMatchSnapshot();
    });
  });

  // Test for the default 'test' account
  test('check content for the default account', () => {
    process.env.AWS_ACCOUNT_ID = 'non_existent_account_id';
    process.env.NODE_ENV = 'production'; // Since the function checks if env is not 'test'

    const defaultContent = getDomainSpecificContent();
    expect(defaultContent).toMatchSnapshot();
  });
});
