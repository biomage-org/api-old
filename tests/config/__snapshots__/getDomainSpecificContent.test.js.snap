// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDomainSpecificContent tests check content for BIOMAGE 1`] = `
Object {
  "adminEmail": "admin@biomage.net",
  "enforcePrivacyPolicyAgreement": true,
  "githubOrganisationName": "biomage-org",
  "moreEmailInfo": "More information about Cellenics can be found at <a href=\\"https://biomage.net\\">biomage.net</a>.<br/><br/>
              If you need help or have any questions, please contact us at hello@biomage.net. <br/><br/>",
  "notificationEmail": "notification@undefined",
}
`;

exports[`getDomainSpecificContent tests check content for HMS 1`] = `
Object {
  "adminEmail": "cellenics_admin@listserv.med.harvard.edu",
  "enforcePrivacyPolicyAgreement": false,
  "githubOrganisationName": "hms-dbmi-cellenics",
  "moreEmailInfo": "",
  "notificationEmail": "alex_pickering@hms.harvard.edu",
}
`;

exports[`getDomainSpecificContent tests check content for the default account 1`] = `
Object {
  "adminEmail": "admin@example.com",
  "enforcePrivacyPolicyAgreement": false,
  "githubOrganisationName": "org",
  "moreEmailInfo": "<div>test hi info</div>",
  "notificationEmail": "some@email.com",
}
`;
